qaframework-bdd-tests/src/test/java/org/openmrs/contrib/qaframework/automation/PhoneNumberValidationSteps.java

```diff
@@ -0,0 +1,128 @@
+import org.openmrs.contrib.qaframework.helper.TestData;
+import org.openmrs.contrib.qaframework.page.RegistrationEditSectionPage;
+
+public class PhoneNumberValidationSteps  extends Steps {
+    
+    private TestData.PatientInfo patient;
+    private RegistrationEditSectionPage registrationEditSectionPage;
+
+    @Before(RunTest.HOOK.SELENIUM_PHONE_NUMBER_VALIDATION)
+    public void setUp() throws Exception {
+        initiateWithLogin();
+        patient = createTestPatient();
+        new TestData.TestVisit(patient.uuid, TestData.getAVisitType(), getLocationUuid(homePage)).create();
+    }
+
+    @After(RunTest.HOOK.SELENIUM_PHONE_NUMBER_VALIDATION)
+    public void tearDown() throws Exception {
+        deletePatient(patient);
+        quit();
+    }  
+    
+    @Given("user goes to find patient record app")
+    public void userGoesToFindPatientApp() {
+        findPatientPage = homePage.goToFindPatientRecord();
+    }
+
+    @When("user searches for patient using patient identifier")
+    public void searchForPatientUsingPatientIdentifier() {
+        findPatientPage.enterPatient(patient.identifier);
+    	findPatientPage.waitForPageToLoad();
+    }
+
+    @And("user clicks on the first patient")
+    public void usersClicksOnFirstPatient() {
+        dashboardPage = findPatientPage.clickOnFirstPatient();
+    }
+
+    @Then("system loads patient dashboard page")
+    public void loadDashboardPage() {
+        assertPage(dashboardPage.waitForPage());
+    }
+    
+    @And("user clicks on show contact link")
+    public void clickOnShowContact() {
+        dashboardPage.clickOnShowContact();
+    }
+
+    @And("user clicks on edit contact info link")
+    public void clickOnEditContact() {
+        registrationEditSectionPage = dashboardPage.clickOnEditContact();
+    }
+
+    @And("user clicks on phone number button")
+    public void userClicksOnPhoneNumber(){
+        registrationEditSectionPage.clickOnPhoneNumberEdit();
+    }
+
+    @And("user clears phone number")
+    public void userClearsPhoneNumber() {
+        registrationEditSectionPage.clearPhoneNumber();
+    }
+
+    @And("user enters an invalid phone number")
+    public void userEntersPhoneNumber() {
+        registrationEditSectionPage.enterPhoneNumber("abcdefg1");
+    }
+
+    @And("user clicks on confirm edit button")
+    public void userClicksOnConfirmEditButton() {
+        registrationEditSectionPage.clickOnConfirmEdit();
+    }
+
+    @Then("system returns a message validator")
+    public void validatePhoneNumber() {
+        assertTrue(registrationEditSectionPage.getInvalidPhoneNumberNotification().contains(("Must be a valid phone number (with +, -, numbers or parentheses)")));
+    }
+
+    @And("user clears the phone number input area")
+    public void userClearPhoneNumber() {
+        registrationEditSectionPage.clearPhoneNumber();
+    }
+
+    @And("user enters the correct phone number")
+    public void userReEntersCorrectPhoneNumber() {
```

Comment:
Drop the `Re` from method name.

Reply:
We are using almost similar method name when i drop Re. it will throw duplicate exception, unless i give it a different method name.

Reply:
What is the name of the other duplicate method name and where is it?

Reply:
@dkayiwa  i renamed them altogether in this [commit ](https://github.com/openmrs/openmrs-contrib-qaframework/pull/370/commits/7f8caa4f8bea9eea3bb33e3895d230fc1f2e3b94)